### PR TITLE
Data Imputation Fix in `erroranalysis`

### PR DESCRIPTION
**Pull request by Advitya17**
Generating the multilabel & OD dashboards from notebooks would result in an uncaught error statement:

Input X contains NaN. Traceback (most recent call last): File "c:\workspace\rai\responsible-ai-toolbox\erroranalysis\erroranalysis\analyzer\error_analyzer.py", line 480, in compute_importances importances = self._compute_error_correlation( File "c:\workspace\rai\responsible-ai-toolbox\erroranalysis\erroranalysis\analyzer\error_analyzer.py", line 519, in _compute_error_correlation return mutual_info_classif( File "c:\Users\agemawat\Anaconda3\envs\rai\lib\site-packages\sklearn\utils_param_validation.py", line 211, in wrapper return func(*args, **kwargs) File "c:\Users\agemawat\Anaconda3\envs\rai\lib\site-packages\sklearn\feature_selection_mutual_info.py", line 493, in mutual_info_classif return _estimate_mi(X, y, discrete_features, True, n_neighbors, copy, random_state) File "c:\Users\agemawat\Anaconda3\envs\rai\lib\site-packages\sklearn\feature_selection_mutual_info.py", line 258, in _estimate_mi X, y = check_X_y(X, y, accept_sparse="csc", y_numeric=not discrete_target) File "c:\Users\agemawat\Anaconda3\envs\rai\lib\site-packages\sklearn\utils\validation.py", line 1147, in check_X_y X = check_array( File "c:\Users\agemawat\Anaconda3\envs\rai\lib\site-packages\sklearn\utils\validation.py", line 959, in check_array _assert_all_finite( File "c:\Users\agemawat\Anaconda3\envs\rai\lib\site-packages\sklearn\utils\validation.py", line 124, in _assert_all_finite _assert_all_finite_element_wise( File "c:\Users\agemawat\Anaconda3\envs\rai\lib\site-packages\sklearn\utils\validation.py", line 173, in _assert_all_finite_element_wise raise ValueError(msg_err) ValueError: Input X contains NaN.

This was because the imputer in `erroranalysis` was not replacing nans due to non-numeric dtype. This PR enforces a numeric dtype for successful imputation and to remove this uncaught exception.

Copilot:
This pull request to the `erroranalysis/erroranalysis` repository adds code to convert the `input_data` array to the `float` data type in the `compute_importances` method of the `ErrorAnalyzer` class. This change ensures error-free calculation and imputation of numerical data types.

* <a href="diffhunk://#diff-2028ff518d9b0e9719fd48a960cc1184cb8ca83990fb9270f590340a927f292aR479-R480">`erroranalysis/erroranalysis/analyzer/error_analyzer.py`</a>: Added code to convert `input_data` array to `float` data type in the `compute_importances` method of the `ErrorAnalyzer` class.

## Checklist

- [ ] I have added screenshots above for all UI changes.
- [ ] I have added e2e tests for all UI changes.
- [ ] Documentation was updated if it was needed.

### CODE REVIEW

File: erroranalysis/erroranalysis/analyzer/error_analyzer.py [hunk 1]
```diff
@@ -476,6 +476,8 @@ def compute_importances(self, error_correlation_method=MUTUAL_INFO):
                                                      IMPORTANCES_THRESHOLD)
             input_data = input_data[indexes]
             diff = diff[indexes]
+        # For numerical data types to make calculation/imputation error-free
+        input_data = input_data.astype(float)
```
Comment (gaugup) [inline erroranalysis/erroranalysis/analyzer/error_analyzer.py:480]: Looks like this converts all column types to float type. What about data types like integer and strings?

**Review by gaugup — COMMENTED** [review]
Could you add a test case for this scenario.

File: erroranalysis/erroranalysis/analyzer/error_analyzer.py (hunk 1 above)
Comment (imatiach-msft) [inline erroranalysis/erroranalysis/analyzer/error_analyzer.py:480]: can you put this inside the catch then?

Comment (Advitya17) [reply erroranalysis/erroranalysis/analyzer/error_analyzer.py:480]: `input_data` encodes all values into numericals it seems:

Example with the metadata for the fridge dataset, `input_data` is:

[[104.43172731850086 0.0 1 0.0 2 244 0.0 31385.0 3024 4032 1 4.0]
 [100.46510438294005 0.0 1 0.0 2 244 0.0 32625.0 3024 4032 1 3.0]
  [102.94642036826406 0.0 1 0.0 2 244 0.0 32776.0 3024 4032 1 0.0]
  [103.5999867603074 0.0 1 0.0 2 244 0.0 31538.0 3024 4032 1 2.0]
  [99.57597176334652 0.0 1 0.0 2 232 0.0 nan 3024 4032 1 8.0]
  [99.52725370661243 0.0 1 0.0 2 244 0.0 31521.0 3024 4032 1 5.0]
  [105.19050714040694 0.0 1 0.0 2 244 0.0 31914.0 3024 4032 1 6.0]
  [99.96797699503111 0.0 1 0.0 2 232 0.0 nan 3024 4032 1 9.0]
  [105.55769296350459 0.0 1 0.0 2 244 0.0 30998.0 3024 4032 1 7.0]
  [109.44899007223657 0.0 1 0.0 2 244 0.0 26287.0 3024 4032 1 1.0]]
  
  and the nans are currently imputed with zeros as per existing logic. The code change just ensures the existing imputation is successful and avoids the uncaught exception.

**Review by imatiach-msft — APPROVED** [review]

**Pull request merged**